01-strings-tasks

troinoiOdekoloN opened the pull request.

troinoiOdekoloN committed:
Update 01-strings-tasks.js

troinoiOdekoloN committed:
Update 01-strings-tasks.js

troinoiOdekoloN committed:
Update 01-strings-tasks.js

Geass1000 commented on task/01-strings-tasks.js:
```diff
@@ -201,10 +196,11 @@ function extractEmails(str) {
  */
 function getRectangleString(width, height) {
-    throw new Error('Not implemented');
+    return '┌'+'─'.repeat(width-2)+'┐'+'\n'+
```
Use more variables.

Geass1000 commented on task/01-strings-tasks.js:
```diff
@@ -221,7 +217,18 @@ function getRectangleString(width, height) {
+  var newString = "";
+  for(var i = 0; i < str.length; i++){
+    if(alphabet.indexOf(str[i]) !== -1){
+      newString += ROTalphabet[alphabet.indexOf(str[i])];
```
Move `alphabet.indexOf(str[i])` to separate variable.

Geass1000 replied:
JFYI: use only variables or constant numbers in `[ ... ]` operator.

Geass1000 commented on task/01-strings-tasks.js:
```diff
@@ -221,7 +217,18 @@ function getRectangleString(width, height) {
+  var ROTalphabet = 'NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm';
+  var newString = "";
+  for(var i = 0; i < str.length; i++){
+    if(alphabet.indexOf(str[i]) !== -1){
```
You can use the ternary operator for this case. Code will be better.

Geass1000 commented on task/01-strings-tasks.js:
```diff
@@ -238,53 +245,59 @@ function encodeToRot13(str) {
  */
 function isString(value) {
-    throw new Error('Not implemented');
+  if (typeof value === 'string' || value instanceof String){
```
`typeof value === 'string' || value instanceof String` return logical result. You don't need to return `true/false` value directly.

Geass1000 commented on task/01-strings-tasks.js:
```diff
@@ -238,53 +245,59 @@ function encodeToRot13(str) {
  */
 function getCardId(value) {
-    throw new Error('Not implemented');
+  return ['A♣','2♣','3♣','4♣','5♣','6♣','7♣','8♣','9♣','10♣','J♣','Q♣','K♣',
```
Use more variables.

troinoiOdekoloN committed:
fixed mistakes in strings-tasks

getRectangleString & getCardId will fix later. encoding of special characters was failed
